### ISSUE TITLE
Some callbacks for debugging and visualization

### ISSUE THREAD

**Issue by awaelchli** [PyTorchLightning/pytorch-lightning-bolts#194]
## 🚀 Feature

Do you have interest in adding any of these callbacks here? 
https://github.com/awaelchli/pytorch-lightning-snippets
Can also make an issue for each one, just wanted to know first if there is a desire for more callbacks in PL bolts?

[TrainingDataMonitor](https://github.com/awaelchli/pytorch-lightning-snippets#monitoring-training-data): logs histogram of what goes into training_step
[ModuleDataMonitor](https://github.com/awaelchli/pytorch-lightning-snippets#monitoring-in--and-outputs-of-network-layers): logs histogram of what goes into each submodule in LightningModule
[BatchNormVerification](https://github.com/awaelchli/pytorch-lightning-snippets#normalization-layers-and-biases): checks (heuristic) if you have a bias before a normalization layer
[BatchGradientVerification](https://github.com/awaelchli/pytorch-lightning-snippets#mixing-data-across-the-batch-dimension): checks that your network does not mix data across batch dim.

### Motivation

- Have more callbacks in PL bolts
- useful for debugging and visualization of training

### Additional context

Happy to help maintain it and adapt to your needs.

**Comment by nateraw** [comment]
These are 🔥 . Let's do it.

**Comment by TylerYep** [comment]
@awaelchli Are you still planning on merging all of these? I figure since you already wrote the code, the verifiers might as well be merged in  :)

**Comment by awaelchli** [comment]
@TylerYep oh yes, thanks for the reminder. 
Not sure how useful the batch norm verification is in general but the gradient check definitely, that's awesome!
I will prepare a PR
